test/tests/functional/pbs_multi_sched.py

```diff
@@ -0,0 +1,601 @@
+        ret = self.server.schedulers['sc5'].start(sched_port='15055')
+        self.server.schedulers['sc5'].log_match(
+            "Scheduler does not contain a partition",
+            max_attempts=10, starttime=self.server.ctime)
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'partition': 'P3'}, id="sc5")
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'Scheduling': 'True'}, id="sc5")
+        self.server.expect(SCHED, {'state': 'idle'}, id='sc5', max_attempts=10)
+        a = {'resources_available.ncpus': 100}
+        self.server.manager(MGR_CMD_SET, NODE, a, id='vnode[2]', expect=True)
+        for _ in xrange(100):
+            j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3'})
+            self.server.submit(j)
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'True'}, id="sc5")
+        self.server.manager(MGR_CMD_LIST, SCHED, id="sc5")
+        self.assertEqual(sched.attributes['state'], 'scheduling')
+        self.du.rm(path=os.path.join(
+            pbs_home, 'sched_priv_sc5'), recursive=True)
+        self.du.rm(path=os.path.join(
+            pbs_home, 'sched_logs_sc5'), recursive=True)
+
+    def test_resource_sched_reconfigure(self):
+        """
+        Test all schedulers will reconfigure while creating,
+        setting or deleting a resource
+        """
+        t = int(time.time())
+        self.server.manager(MGR_CMD_CREATE, RSC, id='foo')
+        sched_list = self.server.schedulers.keys()
+        for name in sched_list:
+            self.server.schedulers[name].log_match(
+                "Scheduler is reconfiguring",
+                max_attempts=10, starttime=t)
+        t = int(time.time())
+        attr = {ATTR_RESC_TYPE: 'long'}
+        self.server.manager(MGR_CMD_SET, RSC, attr, id='foo')
+        for name in sched_list:
+            self.server.schedulers[name].log_match(
+                "Scheduler is reconfiguring",
+                max_attempts=10, starttime=t)
+        t = int(time.time())
+        self.server.manager(MGR_CMD_DELETE, RSC, id='foo')
+        for name in sched_list:
+            self.server.schedulers[name].log_match(
+                "Scheduler is reconfiguring",
+                max_attempts=10, starttime=t)
+
+    def test_remove_partition_sched(self):
+        """
+        Test that removing all the partitions from a scheduler.
+        """
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'partition': (DECR, 'P1')}, id="sc1")
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'partition': (DECR, 'P4')}, id="sc1")
+        log_msg = "Scheduler does not contain a partition"
+        self.server.schedulers['sc1'].log_match(log_msg, max_attempts=10,
+                                                starttime=self.server.ctime)
+        self.server.manager(MGR_CMD_UNSET, SCHED, 'partition',
+                            id="sc2")
```

Comment:
After testing removing of partitions one by one of sc1 and testing log message , I am testing removal of partition of scheduler sc2 by unsetting scheduler attribute . I am not doing log_match because unsetting sched attr will not send any signal back to scheduler and scheduler log wont show that scheduler does not contain partition.

Reply:
@visheshh 
I have not looked into the complete context which you were discussing but just wanted to update that unsetting a few sched attributes like sched_log and sched_priv will trigger a command SCH_ATTRS_CONFIGURE to the scheduler.

Reply:
@suresh-thelkar If any unsetting partition from a scheduler can also send sch_attrs_configure to the scheduler , Scheduler does not contain a partition might appear in sched_logs too.